### DISCUSSION TITLE
http().sendRequest How to set timeout

### DISCUSSION THREAD

**Discussion by M00nBack** [PortSwigger/burp-extensions-montoya-api#87] [General]
http().sendRequest How to set timeout

**Comment by Hannah-PortSwigger** [comment]
Hi. TImeouts should be set at the Burp Project Settings level.

You can find these options under "Settings > Network > Connections > Timeouts".

**Reply by M00nBack** [reply]
Can't I set a timeout on the http().sendRequest() method in montoya-api? I want to set different timeouts for different requests.

**Reply by Hannah-PortSwigger** [reply]
Hi. Just to verify, are you looking to set a limit on the timeout for a response, or are you looking to set timeouts on a different aspect of your request/connection?

**Reply by M00nBack** [reply]
Yes, I am writing a plug-in that uses the http().sendRequest() method in montoya-api to send requests and the waiting time is very long. I don’t want to modify the Burp project settings, and I did not find the parameter to set the sendRequest method timeout.

**Reply by Hannah-PortSwigger** [reply]
Thanks for the clarification! This functionality is not currently possible with the Montoya API. However, we can raise a feature request to add this functionality at a later date.

In the meantime, your only option is to adjust the timeout settings on a project level.

Please let us know if you need any further assistance.
